projects/angular-jwt/src/lib/jwthelper.service.ts

```diff
@@ -77,12 +76,61 @@ export class JwtHelperService {
     );
   }
 
-  public decodeToken<T = any>(token: string = this.tokenGetter()): T {
-    if (!token || token === "") {
+  public decodeToken(
+    token: string | Promise<string> = this.tokenGetter()
+  ): any | Promise<any> {
```

Comment:
I think you forgot the generics that was here previously

Reply:
@piotaixr thanks! Good spot. Improved the generics and type signatures